### PR TITLE
typescript-angular: Cancel edits using Escape key

### PR DESCRIPTION
**Pull request by dmethvin**
Ref #789 

While building I noticed that the `.js` files generated by the `.ts` files were committed to the repo. When the `npm run compile` command is used, it generates the transpiled `Application.js` from *all* the `.ts` files so these files are not needed and are not refreshed by `npm run compile` anyway. So I removed them in a separate commit.

### CODE REVIEW

**Comment by samccone** [comment]
:bow: amazing work @dmethvin 

:+1: from me

File: examples/typescript-angular/js/Application.js [hunk 1]
```diff
@@ -53,6 +53,28 @@ var todos;
 })(todos || (todos = {}));
 /// <reference path='../_all.ts' />
 var todos;
+(function (todos) {
+    'use strict';
+    var ESCAPE_KEY = 27;
```
Comment (samccone) [inline examples/typescript-angular/js/Application.js]: nit: can we const here?

Comment (dmethvin) [inline examples/typescript-angular/js/Application.js]: Why yes, yes you can. I forgot that TypeScript supported it.

Comment (dmethvin) [inline examples/typescript-angular/js/Application.js]: Although the change will be reflected in `TodoEscape.ts` down the diff, this is the compiled file.

File: examples/typescript-angular/js/directives/TodoEscape.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	export function todoEscape(): ng.IDirective {
+		return {
+			link: ($scope: ng.IScope, element: JQuery, attributes: any) => {
+				element.bind('keydown', (event) => {
+					if (event.keyCode === ESCAPE_KEY) {
+						$scope.$apply(attributes.todoEscape);
+					}
+				});
+
+				$scope.$on('$destroy', () => { element.unbind('keydown'); });
```
Comment (samccone) [inline examples/typescript-angular/js/directives/TodoEscape.ts]: yay no leaks :)

File: examples/typescript-angular/js/Application.js (hunk 1 above)
Comment (samccone) [inline examples/typescript-angular/js/Application.js]: yep sorry commented on the wrong file :)

**Comment by samccone** [comment]
Landed this PR manually

https://github.com/tastejs/todomvc/commit/7d6eb8964e13a17e8e45260f1c25b506386e7da9
https://github.com/tastejs/todomvc/commit/77201888909aca20d9e7264ff35f7ff22cd82ecd

Thanks a lot @dmethvin :dancer:

**Pull request closed without merging**